### PR TITLE
Graceful fails on httpRequest

### LINKED ISSUE #703: `response` object is undefined when calling Parse.Cloud.httpRequest with nonsense url
I'm getting an exception whenever I call Parse.Cloud.httpRequest with a nonsense URL. Ideally it should put something sensible in the error callback instead of crashing.

#### Environment Setup
parse-server v2.1.3

#### Steps to reproduce

```js
          var promise = Parse.Cloud.httpRequest({
            method: "GET",
            url: "https://www.blah blahblah",
          });
```

#### Logs/Trace
```
TypeError: Cannot read property 'statusCode' of undefined
    at Request._callback (cloudcode/dist/node_modules/parse-server/lib/httpRequest.js:46:35)
    at self.callback (cloudcode/dist/node_modules/parse-server/node_modules/request/request.js:199:22)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
```

### PR DESCRIPTION
**Pull request by flovilmart**
fixes  #703

### CODE REVIEW

File: src/cloud-code/httpRequest.js [hunk 1]
```diff
@@ -36,6 +36,12 @@ module.exports = function(options) {
   options.followRedirect = options.followRedirects == true;
   
   request(options, (error, response, body) => {
+    if (error) {
+      if (callbacks.error) {
+        callbacks.error(error);
+      }
+      return promise.reject(error);
```
Comment (nlutsenko) [inline src/cloud-code/httpRequest.js]: Nit: `Promise.reject()`

File: src/cloud-code/httpRequest.js [hunk 2]
```diff
@@ -46,7 +52,7 @@ module.exports = function(options) {
       httpResponse.data = JSON.parse(response.body);
     } catch (e) {}
     // Consider <200 && >= 400 as errors 
-    if (error || httpResponse.status <200 || httpResponse.status >=400) {
+    if (httpResponse.status <200 || httpResponse.status >=400) {
```
Comment (nlutsenko) [inline src/cloud-code/httpRequest.js]: Nit: Space before-after `<` and `>=`.

**Comment by nlutsenko** [comment]
LGTM, 2 nits.

File: src/cloud-code/httpRequest.js (hunk 1 above)
Comment (flovilmart) [inline src/cloud-code/httpRequest.js]: that won't work as we're in a callback, promise is a `new Parse.Promise()`

**Pull request merged**